diff --git a/CMakeLists.txt b/CMakeLists.txt
index c9d8b5f..48b5445 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -333,7 +333,7 @@ configure_file(
         ${PROJECT_SOURCE_DIR}/cmake/modules/SRSLTE_install_configs.sh.in
         ${CMAKE_BINARY_DIR}/srslte_install_configs.sh
 )
-install(PROGRAMS ${CMAKE_BINARY_DIR}/srslte_install_configs.sh DESTINATION ${RUNTIME_DIR})
+#install(PROGRAMS ${CMAKE_BINARY_DIR}/srslte_install_configs.sh DESTINATION ${RUNTIME_DIR})
 
 # Disables the project to build when calling "make install"
 set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY TRUE)
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index fa4741d..2c782c9 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -29,6 +29,6 @@ add_subdirectory(test)
 ########################################################################
 # Install library headers
 ########################################################################
-INSTALL(  DIRECTORY include/
-          DESTINATION "${INCLUDE_DIR}"
-          FILES_MATCHING PATTERN "*.h" )
+#INSTALL(  DIRECTORY include/
+#          DESTINATION "${INCLUDE_DIR}"
+#          FILES_MATCHING PATTERN "*.h" )
diff --git a/lib/include/srslte/CMakeLists.txt b/lib/include/srslte/CMakeLists.txt
index 8bfa689..2ca8d74 100644
--- a/lib/include/srslte/CMakeLists.txt
+++ b/lib/include/srslte/CMakeLists.txt
@@ -23,6 +23,6 @@ configure_file(
     ${CMAKE_CURRENT_BINARY_DIR}/version.h
 )
 
-INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/version.h
-        DESTINATION "${INCLUDE_DIR}/srslte"
-)
+#INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/version.h
+#        DESTINATION "${INCLUDE_DIR}/srslte"
+#)
diff --git a/srsenb/CMakeLists.txt b/srsenb/CMakeLists.txt
index d7120f6..8210da8 100644
--- a/srsenb/CMakeLists.txt
+++ b/srsenb/CMakeLists.txt
@@ -52,7 +52,7 @@ add_subdirectory(test)
 ########################################################################
 # Default configuration files
 ########################################################################
-install(FILES enb.conf.example DESTINATION ${DATA_DIR})
-install(FILES drb.conf.example DESTINATION ${DATA_DIR})
-install(FILES rr.conf.example DESTINATION ${DATA_DIR})
-install(FILES sib.conf.example DESTINATION ${DATA_DIR})
\ No newline at end of file
+#install(FILES enb.conf.example DESTINATION ${DATA_DIR})
+#install(FILES drb.conf.example DESTINATION ${DATA_DIR})
+#install(FILES rr.conf.example DESTINATION ${DATA_DIR})
+#install(FILES sib.conf.example DESTINATION ${DATA_DIR})
diff --git a/srsepc/CMakeLists.txt b/srsepc/CMakeLists.txt
index 6ebe471..ba12adc 100644
--- a/srsepc/CMakeLists.txt
+++ b/srsepc/CMakeLists.txt
@@ -51,7 +51,7 @@ add_subdirectory(src)
 ########################################################################
 # Default configuration files
 ########################################################################
-install(FILES epc.conf.example DESTINATION ${DATA_DIR})
-install(FILES mbms.conf.example DESTINATION ${DATA_DIR})
-install(FILES user_db.csv.example DESTINATION ${DATA_DIR})
-install(PROGRAMS srsepc_if_masq.sh DESTINATION ${RUNTIME_DIR})
\ No newline at end of file
+#install(FILES epc.conf.example DESTINATION ${DATA_DIR})
+#install(FILES mbms.conf.example DESTINATION ${DATA_DIR})
+#install(FILES user_db.csv.example DESTINATION ${DATA_DIR})
+#install(PROGRAMS srsepc_if_masq.sh DESTINATION ${RUNTIME_DIR})
diff --git a/srsue/CMakeLists.txt b/srsue/CMakeLists.txt
index 9b62972..1b493ef 100644
--- a/srsue/CMakeLists.txt
+++ b/srsue/CMakeLists.txt
@@ -48,4 +48,4 @@ add_subdirectory(test)
 ########################################################################
 # Default configuration files
 ########################################################################
-install(FILES ue.conf.example DESTINATION ${DATA_DIR})
\ No newline at end of file
+#install(FILES ue.conf.example DESTINATION ${DATA_DIR})
